### PR TITLE
Fix ObservationTable select time

### PR DESCRIPTION
**Pull request by registerrier**
**Description**

This pull request corrects an improper behavior in `ObservationTable.select_time` reported in #2613.
This PR modifies the behavior of `ObservationTable.select_time` introducing `time_start` and `time_stop` properties. This creates a behavior closer to `EventList.select_time`.
Here we check for overlap, since an observation can be fully or partly enclosed in the time bin and we leave the option to the user to select one case or the other. 

The test no longer use a table with absolute times stored. `ObservationTable` now assumes only a table with time reference and elapse time as all other time tables in gammapy. 

**Dear reviewer**

The PR is ready for review. The new code is tested.

### CODE REVIEW

File: gammapy/data/tests/test_obs_table.py [hunk 1]
```diff
@@ -259,18 +259,26 @@ def test_select_time_box():
...
     value_range = Time(["2012-01-01T01:00:00", "2012-01-01T02:00:00"])
     selection = dict(type="time_box", time_range=value_range)
     selected_obs_table = obs_table_time.select_observations(selection)
-    time_start = selected_obs_table["TSTART"]
-    assert (value_range[0] < time_start).all()
-    assert (time_start < value_range[1]).all()
+    time_start = selected_obs_table.time_start
+    time_stop = selected_obs_table.time_stop
 
+    assert (value_range[0] < time_start).all() & (time_stop < value_range[1]).all()
```
Comment (cdeil) [inline gammapy/data/tests/test_obs_table.py]: Typical case of re-implementing the method in the test. Should be changed to a simple small test case and direct assert on output. But OK, that's old baggage that was already here.

File: gammapy/data/tests/test_obs_table.py [hunk 2]
```diff
@@ -259,18 +259,26 @@ def test_select_time_box():
...
+
+    # Test selection with partial overlap of observations and value_range
+    selection_partial = dict(type="time_box", time_range=value_range, partial_overlap=True)
+    selected_obs_table = obs_table_time.select_observations(selection_partial)
+    time_start = selected_obs_table.time_start
+    time_stop = selected_obs_table.time_stop
+
+    assert (value_range[1] > time_start).all() & (time_stop > value_range[0]).all()
```
Comment (cdeil) [inline gammapy/data/tests/test_obs_table.py]: Another copy of the time select implementation code in the test. Sigh.

**Comment by cdeil** [comment]
I changed the title to "Fix ObservationTable select time". IMO it's nice to have most PRs start with "Fix", "Add", "Remove" or "Improve" - then PR titles in the changelog can be parsed quickly, and users can see where bugs were fixed.

File: gammapy/data/tests/test_obs_table.py (hunk 1 above)
Comment (registerrier) [reply gammapy/data/tests/test_obs_table.py]: Yes that is not very good. But with random values it is difficult to do differently. Indeed the tests should be completely refactored.

**Comment by registerrier** [comment]
Having a hard-coded selection on `TSTART` as before is not very nice. So the fully contained observation seems a more expected behavior.
I have introduced `partial_overlap` also because the `inverted` option is meaningful if you have this option.  

I propose to merge for now, at least to solve #2613. All tests should be refactored at some point and this code clarified. I open an issue for that. To be sorted out hopefully for 1.0.

**Pull request merged**